src/Sylius/Bundle/ApiBundle/ApiPlatform/Metadata/Merger/LegacyResourceMetadataMerger.php

```diff
@@ -23,6 +23,10 @@ public function merge(array $oldMetadata, array $newMetadata): array
 
         foreach ($newMetadata as $key => $value) {
             if ('properties' === $key) {
+                if ($value === null) {
```

Comment:
maybe adjusting \Sylius\Bundle\ApiBundle\ApiPlatform\Metadata\MergingXmlExtractor::buildResource properties is better?